src/main/java/com/receipts/on/ReceiptRepository.java

```diff
@@ -30,17 +38,45 @@ public ReceiptRepository(DB db) {
         return prescriptions;
     }
 
-    public Prescription find(String id) {
+    public Optional<Prescription> find(Long id) {
         DBObject query = new BasicDBObject();
-        query.put("prescriptionId", new ObjectId(id));
+        query.put("prescriptionId", new ObjectId(String.valueOf(id)));
         DBObject dbObject = collection.findOne(query);
-        return deserializePrescription(dbObject);
+
+        if (dbObject == null)
+            return Optional.empty();
+
+        return Optional.of(deserializePrescription(dbObject));
     }
- 
-    public void createNew(String body) {
-        Prescription prescription = new Gson().fromJson(body, Prescription.class);
-        DBObject doc = serialize(prescription);
-        collection.insert(doc);
+
+    public boolean update(Prescription prescription) {
+        if (!exists(prescription))
+            return false;
+
+        DBObject query = new BasicDBObject();
+        query.put("prescriptionId", new ObjectId(String.valueOf(prescription.prescriptionId())));
+        collection.update(query, serialize(prescription));
+        return true;
+    }
+
+    public boolean delete(Long id) {
```

Comment:
I don't think that you can actually delete the prescription. What about history?